src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
+    );
 
-        return {
-            switchStyle: switchStyles,
-            props:
-                version && version.major === 0 && version.minor <= 56
-                    ? {
-                          onTintColor,
-                          thumbTintColor,
-                      }
-                    : Platform.OS === 'web'
-                    ? {
-                          activeTrackColor: onTintColor,
-                          thumbColor: thumbTintColor,
-                          activeThumbColor: checkedColor,
-                      }
-                    : {
-                          thumbColor: thumbTintColor,
-                          trackColor: {
-                              true: onTintColor,
-                              false: onTintColor,
-                          },
-                      },
-        };
-    }, [componentStyles, disabled, value]);
+    useEffect(() => {
+        Animated.timing(toggleMarginAnimation, {
+            toValue: value ? 1 : 0,
+            duration: 300,
+            easing: Easing.bezier(0.175, 0.885, 0.32, 1.275),
+            useNativeDriver: false,
+        }).start();
+        Animated.timing(toggleSizeAnimation, {
+            toValue: value ? 1 : 0,
+            duration: 250,
+            easing: Easing.ease,
+            useNativeDriver: false,
+        }).start();
+    }, [value, toggleMarginAnimation, toggleSizeAnimation]);
 
-    return (
-        <NativeSwitch
-            ref={ref}
-            value={value}
-            disabled={disabled}
-            onValueChange={disabled ? undefined : onValueChange}
-            style={switchStyle}
-            {...props}
-            {...rest}
-        />
+    const thumbOverlay = useMemo(
+        () => ({
+            position: 'absolute',
+            left:
+                value && pressed
+                    ? size * -0.21
+                    : value || unCheckedIcon
+                    ? size * -0.28
+                    : !unCheckedIcon && !pressed
+                    ? size * -0.36
+                    : size * -0.16,
+            right: 0,
+            height: size * 1.2,
+            width: size * 1.2,
+            borderRadius: (size * 1.2) / 2,
+            backgroundColor: pressed
+                ? 'rgba(0, 0, 0, 0.15)'
+                : focused
+                ? 'rgba(0, 0, 0, 0.1)'
+                : 'rgba(0, 0, 0, 0.05)',
+            display: hovered || focused || pressed ? 'flex' : 'none',
+        }),
+        [focused, hovered, pressed, size, unCheckedIcon, value],
     );
-};
 
-type CustomProp = { checkedColor?: string; thumbTintColor?: string; onTintColor?: string };
+    const { parentViewStyle, buttonStyle, toggleStyle, toggleWheelStyle, overlayStyle, iconStyle } =
+        useMemo(() => {
+            const { button, toggle, toggleWheel, icon } = componentStyles;
+            return {
+                buttonStyle: [button, size && { borderRadius: size / 2 }],
```

Comment:
Can we have a better names here?
buttonStyle should be switchStyle
parentViewStyle should be thumbContainerStyle
toggleStyle and switchStyle should be combined.
toggleWheelStyle should be thumbStyle
overlayStyle should be thumbOverlayStyle